From 0d8679084d5f096c4a233fdf2abd7c5367ca7a43 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Tue, 11 Apr 2023 12:30:34 +0300
Subject: [PATCH 5/8] net: ethernet: ti: icssg_prueth: fix kernel oops in probe
 when phy fails

During prueth_probe return error if emac_phy_connect returns an error.
Otherwise the phy_attached_info function called next will trigger oops.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 drivers/net/ethernet/ti/icssg_prueth.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/ti/icssg_prueth.c b/drivers/net/ethernet/ti/icssg_prueth.c
index b4beb2239a69..6c09b99d6335 100644
--- a/drivers/net/ethernet/ti/icssg_prueth.c
+++ b/drivers/net/ethernet/ti/icssg_prueth.c
@@ -3396,7 +3396,10 @@ static int prueth_probe(struct platform_device *pdev)
 					  prueth->emac[PRUETH_MAC0]->ndev);
 		prueth->registered_netdevs[PRUETH_MAC0] = prueth->emac[PRUETH_MAC0]->ndev;
 
-		emac_phy_connect(prueth->emac[PRUETH_MAC0]);
+		ret = emac_phy_connect(prueth->emac[PRUETH_MAC0]);
+		if (ret)
+			return ret;
+
 		/* Get attached phy details */
 		phy_attached_info(prueth->emac[PRUETH_MAC0]->phydev);
 
@@ -3414,7 +3417,10 @@ static int prueth_probe(struct platform_device *pdev)
 
 		prueth->registered_netdevs[PRUETH_MAC1] = prueth->emac[PRUETH_MAC1]->ndev;
 
-		emac_phy_connect(prueth->emac[PRUETH_MAC1]);
+		ret = emac_phy_connect(prueth->emac[PRUETH_MAC1]);
+		if (ret)
+			return ret;
+
 		/* Get attached phy details */
 		phy_attached_info(prueth->emac[PRUETH_MAC1]->phydev);
 	}
-- 
2.35.3

